examples/all-clusters-app/all-clusters-common/src/WhmDelegateImpl.cpp

```diff
@@ -0,0 +1,461 @@
+{
+    if (mBoostState != boostState)
+    {
+        mBoostState = boostState;
+
+        MatterReportingAttributeChangeCallback(mEndpointId, WaterHeaterManagement::Id, Attributes::BoostState::Id);
+    }
+}
+
+/**
+ * @brief Handles the boost command
+ *
+ * Upon receipt, the Water Heater SHALL transition into the BOOST state, which SHALL cause the water in the tank (or
+ * the TargetPercentage of the water, if included) to be heated towards the set point (or the TemporarySetpoint, if
+ * included), which in turn may cause a call for heat, even if the mode is OFF, or is TIMED and it is during one of
+ * the Off periods.
+ */
+Status WaterHeaterManagementDelegate::HandleBoost(uint32_t durationS, Optional<bool> oneShot, Optional<bool> emergencyBoost,
+                                                  Optional<int16_t> temporarySetpoint, Optional<Percent> targetPercentage,
+                                                  Optional<Percent> targetReheat)
+{
+    Status status = Status::Success;
+
+    ChipLogProgress(AppServer, "HandleBoost");
+
+    // Keep track of the boost command parameters
+    mBoostOneShot           = oneShot;
+    mBoostEmergencyBoost    = emergencyBoost;
+    mBoostTemporarySetpoint = temporarySetpoint;
+    mBoostTargetPercentage  = targetPercentage;
+    mBoostTargetReheat      = targetReheat;
+
+    mBoostTargetTemperatureReached = false;
+
+    // If a timer is running, cancel it so we can start a new boost command with the new duration
+    if (mBoostState == BoostStateEnum::kActive)
+    {
+        DeviceLayer::SystemLayer().CancelTimer(BoostTimerExpiry, this);
+    }
+
+    CHIP_ERROR err = DeviceLayer::SystemLayer().StartTimer(System::Clock::Seconds32(durationS), BoostTimerExpiry, this);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(AppServer, "HandleBoost: Unable to start a Boost timer: %" CHIP_ERROR_FORMAT, err.Format());
+
+        // Not a lot we can do -> just set the boost state to inactive
+        SetBoostState(BoostStateEnum::kInactive);
+
+        return Status::Failure;
+    }
+
+    // Now running a boost command
+    SetBoostState(BoostStateEnum::kActive);
+
+    if (mpWhmManufacturer != nullptr)
+    {
+        status = mpWhmManufacturer->BoostCommandStarted(durationS, oneShot, emergencyBoost, temporarySetpoint, targetPercentage,
+                                                        targetReheat);
+    }
+    else
+    {
+        status = Status::InvalidInState;
+        ChipLogError(AppServer, "HandleBoost: mpWhmManufacturer == nullptr");
+    }
+
+    if (status == Status::Success)
+    {
+        // See if the heat needs to be turned on or off as a result of this boost command
+        status = CheckIfHeatNeedsToBeTurnedOnOrOff();
+    }
+
+    return status;
+}
+
+void WaterHeaterManagementDelegate::BoostTimerExpiry(System::Layer * systemLayer, void * delegate)
+{
+    WaterHeaterManagementDelegate * dg = reinterpret_cast<WaterHeaterManagementDelegate *>(delegate);
```

Comment:
Why are we not using a static_cast here? are we expecting to receive pointers or different types?

Reply:
Code updated